### PR TITLE
[eos-noon] introducing ctest: test runner by cmake toolset

### PR DESCRIPTION
**Pull request by asiniscalchi**

### CODE REVIEW

File: libraries/chain/test/name_test.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#include <boost/test/unit_test.hpp>
+
+#include "eosio/chain/name.hpp"
+
+BOOST_AUTO_TEST_SUITE(name_test)
+
+BOOST_AUTO_TEST_CASE(dafault_to_string)
```
Comment (jgiszczak) [inline libraries/chain/test/name_test.cpp]: dafult_to_string -> default_to_string

Comment (asiniscalchi) [reply libraries/chain/test/name_test.cpp]: I don't get it

**Comment by asiniscalchi** [comment]
image

**Comment by asiniscalchi** [comment]
parallel testing is also supported:
ctest -jN

File: libraries/chain/test/name_test.cpp (hunk 1 above)
Comment (wanderingbort) [reply libraries/chain/test/name_test.cpp]: "default" is misspelled as "dafault".  Unless, this is a trendy way of saying "the fault to string" 😄 

Either way, we should clean it up.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -61,9 +61,17 @@ endif()
...
 #configure_file(${CMAKE_CURRENT_SOURCE_DIR}/eosd_run_mongodb_test.sh ${CMAKE_CURRENT_BINARY_DIR}/eosd_run_mongodb_test.sh COPYONLY)
 #configure_file(${CMAKE_CURRENT_SOURCE_DIR}/p2p_tests/sync/test.sh ${CMAKE_CURRENT_BINARY_DIR}/p2p_tests/sync/test.sh COPYONLY)
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/run_tests.sh ${CMAKE_CURRENT_BINARY_DIR}/run_tests.sh COPYONLY)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/trans_sync_across_mixed_cluster_test.sh ${CMAKE_CURRENT_BINARY_DIR}/trans_sync_across_mixed_cluster_test.sh COPYONLY)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/distributed-transactions-test.py ${CMAKE_CURRENT_BINARY_DIR}/distributed-transactions-test.py COPYONLY)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/restart-scenarios-test.py ${CMAKE_CURRENT_BINARY_DIR}/restart-scenarios-test.py COPYONLY)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/testUtils.py ${CMAKE_CURRENT_BINARY_DIR}/testUtils.py COPYONLY)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/eosiod_run_test.py ${CMAKE_CURRENT_BINARY_DIR}/eosiod_run_test.py COPYONLY)
+
+if(UNIX AND NOT APPLE)
```
Comment (wanderingbort) [inline tests/CMakeLists.txt]: After talking with some of our CI guys, I don't think it makes sense to exclude these tests from OSX at _build_ time.  We still want travis to skip them for OSX however, individual developers on mac will still want to use this during their daily builds.  

What do you think about moving the selection of tests to `.travis.yml` using [the command line](https://cmake.org/Wiki/CMake/Testing_With_CTest#Running_Individual_Tests)?

File: libraries/chain/test/name_test.cpp (hunk 1 above)
Comment (asiniscalchi) [reply libraries/chain/test/name_test.cpp]: oh lol

File: tests/CMakeLists.txt (hunk 2 above)
Comment (asiniscalchi) [reply tests/CMakeLists.txt]: done

**Pull request merged**